exchange-wrapper/contracts/ExchangeWrapper.sol

```diff
@@ -101,12 +101,18 @@ contract ExchangeWrapper is OwnableUpgradeable {
     function matchExchangeV2(
         LibOrder.Order memory sellOrder,
         bytes memory sellOrderSignature,
-        uint amount
+        uint paymentAmount,
+        uint purchaseAmount
     ) internal {
         LibOrder.Order memory buyerOrder;
         buyerOrder.maker = address(this);
         buyerOrder.makeAsset = sellOrder.takeAsset;
-        buyerOrder.takeAsset = sellOrder.makeAsset;
+        buyerOrder.takeAsset.assetType = sellOrder.makeAsset.assetType;
+        if (buyerOrder.takeAsset.assetType.assetClass == LibAsset.ERC1155_ASSET_CLASS) {
+            buyerOrder.takeAsset.value = purchaseAmount; //for ERC1155 set the exact amount for purchase
+        } else {
+            buyerOrder.takeAsset.value = sellOrder.makeAsset.value; // for ERC721 is always == 1
+        }
```

Comment:
why do we need this if? if purchaseAmount is always present, we can assume that for ERC721 it's going to be 1. or not?

Reply:
I agree, better to use simple:
buyerOrder.takeAsset.value = purchaseAmount;
without any If, "if" don`t save from error if parameter  **purchaseAmount** is incorrect